### ISSUE TITLE
Assertion failed: (!r), function lyd_keyless_list_hash_change

### ISSUE THREAD

**Issue by kwatsen** [CESNET/libyang#867]
Using yanglint 1.1.39.

```
yanglint --verbose --strict --type data  module.yang  document.json
Assertion failed: (!r), function lyd_keyless_list_hash_change, file /Users/kent/GitHub/cesnet/libyang/src/tree_data.c, line 391.
Abort trap: 6
```

Yes, I have a couple `config false` lists, neither of which have a `key` statement.

Playing around with the input document a little, I can sometimes make the error disappear by deleting parts of the `config false` list data, but so far I cannot ascertained why, or how to workaround the issue.

Any clue on what might cause this?

**Comment by fredgan** [comment]
Would you mind providing  the files: module.yang  document.json ?

**Comment by kwatsen** [comment]
> Would you mind providing the files: module.yang document.json ?
> 
> 

I cannot, at least not in their current form, as they contain proprietary information.
I could try to generate a unit test if I had a clue as to what this error might regard...
My current best guess is that a collision is occurring with the hashing routine.  
I say this because, e.g., say there is a config-false list with values:

  "widgets" : {
      "widget" : [
        {
        "a" : { ... }
        "b" : { ... }
        "c" : { ... }
      ]
  }

It "works" if I delete 'c', but it also works if I delete 'a' and 'b'.   So, 'c' by itself isn't the problem, but somehow it is in the presence of 'a' and 'b'.  Does it make any sense to you?

Kent

**Comment by rkrejci** [comment]
Thanks for reporting Kent, it should be fixed in 1.1.44. The problem was connected with maintaining hash tables with keyless lists. They are used only with at least 4 children, so it could be reason why the problem dissappeared sometimes.

**Comment by rkrejci** [comment]
I had to change the fix, so 1.1.48 should be better.
